pkg/controllers/sync/controller.go

```diff
@@ -926,33 +769,40 @@ func (s *SyncController) ensureRemovedOrUnmanaged(ctx context.Context, fedResour
 // each managed resource in member clusters.
 func (s *SyncController) handleDeletionInClusters(
 	ctx context.Context,
-	gvk schema.GroupVersionKind,
-	qualifiedName common.QualifiedName,
+	targetGVK schema.GroupVersionKind,
+	targetQualifiedName common.QualifiedName,
 	deletionFunc func(dispatcher dispatch.UnmanagedDispatcher, clusterName string, clusterObj *unstructured.Unstructured),
 ) (bool, error) {
-	clusters, err := s.informer.GetJoinedClusters()
+	keyedLogger := klog.FromContext(ctx)
+
+	clusters, err := s.fedInformerManager.GetJoinedClusters()
 	if err != nil {
-		return false, errors.Wrap(err, "failed to get a list of clusters")
+		return false, fmt.Errorf("failed to get a list of clusters: %w", err)
+	}
+
+	ftc, exists := s.ftcManager.GetResourceFTC(targetGVK)
```

Comment:
Let's avoid retrieving FTC from `ftcManager` again, and use the one from the `FederatedResource`. We can pass it as an argument to this function in `deleteFromClusters`.